haystack/pipeline.py

```diff
@@ -8,6 +9,8 @@
 from typing import List, Optional, Dict, Union, Any
 import pickle
 import urllib
+import ray
```

Comment:
Let's make it an optional import here. Not every user will need it and with #1070 we will have versions like "haystack[minimal]" which should not require ray